modules/dnn/src/layers/layers_common.cpp

```diff
@@ -55,128 +55,142 @@ std::string makeName(const std::string& str1, const std::string& str2)
-                parameterW = parameterH;
-            }
-            else if (param.size() == 2)
-            {
-                parameterW = param.get<int>(1);
-            }
-            else
-            {
-                return false;
+            for (int i = 0; i < param.size(); i++) {
+                CV_Assert(param.get<int>(i) >= 0);
+                parameter.push_back(param.get<int>(i));
             }
             return true;
         }
         else
         {
-            if(hasDefault)
+            if (hasDefault)
             {
-                parameterH = parameterW = defaultValue;
+                parameter.push_back(defaultValue);
                 return true;
             }
-            else
-            {
-                return false;
-            }
+            return false;
         }
     }
 }
 
-void getKernelSize(const LayerParams &params, int &kernelH, int &kernelW)
+void getKernelSize(const LayerParams &params, std::vector<size_t>& kernel)
 {
-    if(!util::getParameter(params, "kernel", "kernel_size", kernelH, kernelW))
-    {
+    if (!util::getParameter(params, "kernel", "kernel_size", kernel))
         CV_Error(cv::Error::StsBadArg, "kernel_size (or kernel_h and kernel_w) not specified");
-    }
 
-    CV_Assert(kernelH > 0 && kernelW > 0);
+    if (kernel.size() == 1)
+        kernel.resize(2, kernel[0]);
+
+    for (int i = 0; i < kernel.size(); i++)
+        CV_Assert(kernel[i] > 0);
 }
 
-void getStrideAndPadding(const LayerParams &params, int &padT, int &padL, int &padB, int &padR, int &strideH, int &strideW, cv::String& padMode)
+void getStrideAndPadding(const LayerParams &params, std::vector<size_t>& pads_begin, std::vector<size_t>& pads_end,
+                         std::vector<size_t>& strides, cv::String& padMode)
 {
     if (params.has("pad_l") && params.has("pad_t") && params.has("pad_r") && params.has("pad_b")) {
-        padT = params.get<int>("pad_t");
-        padL = params.get<int>("pad_l");
-        padB = params.get<int>("pad_b");
-        padR = params.get<int>("pad_r");
+        CV_Assert(params.get<int>("pad_t") >= 0 && params.get<int>("pad_l") >= 0 &&
+                  params.get<int>("pad_b") >= 0 && params.get<int>("pad_r") >= 0);
+        pads_begin.push_back(params.get<int>("pad_t"));
+        pads_begin.push_back(params.get<int>("pad_l"));
+        pads_end.push_back(params.get<int>("pad_b"));
+        pads_end.push_back(params.get<int>("pad_r"));
     }
     else {
-        util::getParameter(params, "pad", "pad", padT, padL, true, 0);
-        padB = padT;
-        padR = padL;
+        util::getParameter(params, "pad", "pad", pads_begin, true, 0);
+        if (pads_begin.size() < 4)
+            pads_end = pads_begin;
+        else
+        {
+            pads_end = std::vector<size_t>(pads_begin.begin() + pads_begin.size() / 2, pads_begin.end());
+            pads_begin.resize(pads_begin.size() / 2);
+        }
+        CV_Assert(pads_begin.size() == pads_end.size());
```

Comment:
It seems to me that "pad" field contain only begin paddings (at least it is right now). Is there a test with "pad" filled by both begins and ends?